packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.test.js

```diff
@@ -84,11 +88,91 @@ describe(name, () => {
     click(generateBtn);
     expect(onRequestSubmit).toHaveBeenCalled();
 
-    rerender(<APIKeyModal {...defaultProps} apiKey="444-444-444-444" />);
+    rerender(<APIKeyModal {...props} loading />);
+    expect(getByText(props.loadingMessage)).toBeVisible();
+    rerender(<APIKeyModal {...props} apiKey="444-444-444-444" />);
     const copyBtn = getByText('Copy');
     click(copyBtn);
     expect(navigator.clipboard.writeText).toHaveBeenCalledWith(
       '444-444-444-444'
     );
   });
+
+  it('renders with steps setup', () => {
+    const { click } = fireEvent;
+    const { fn } = jest;
+    const onRequestSubmit = fn();
+    const onRequestClose = fn();
+    const customSteps = [
+      {
+        valid: true,
+        content: <input type="text" value="a" placeholder="input a" readOnly />,
+      },
+      {
+        valid: true,
+        content: <input type="text" value="b" placeholder="input b" readOnly />,
+      },
+      {
+        valid: false,
+        content: <input type="text" value="c" placeholder="input c" readOnly />,
+      },
+    ];
+    const props = {
+      ...standardProps,
+      onRequestSubmit,
+      onRequestClose,
+      customSteps,
+      previousStepButtonText: 'Previous step',
+      nextStepButtonText: 'Next step',
+      downloadable: false,
+    };
+
+    const { rerender, getByPlaceholderText, getByText } = render(
+      <APIKeyModal {...props} />
+    );
+
+    // step 1
+    expect(getByPlaceholderText('input a')).toBeVisible();
```

Comment:
On the assumption that only render the current step is rendered you do not need `expect().toBeVisible`. screen.get... and screen.find... throw errors if they fail to locate the target.

Reply:
true. my thinking is i'm not actually targeting the inputs to preform any action on them. the test in question has more to do with testing the flow works and not about interacting with the content, so since i'm not trying to perform any input actions using `toBeVisible` is an easy way to test that clicking the next or previous button moves the user to the correct step and displays the correct content. it also makes the tests easier to read (in my opinion).